mantidproject/mantid
# StartLiveDataDialog does not set start time correctly.

**Description of work.**

In  **StartLiveDataDialog** , if a user sets the *Starting Time* to an arbitrary value, as soon as algorithm is execute, an error message is issued and execute quits as failure.  The error message is 

```
Error in execution of algorithm StartLiveData:  Error interpreting string '' as a date/time.
```

The reason is that all the property value set by `setProperty` before `accept()` is called will be replaced by the value recorded in `AlgorithmDialog::m_propertyValueMap` in method `AlgorithmDialog::accept()`, which is called to execute the algorithm.

The solution is to replace `setProperty` by `storePropertyValue`.

This is to resolve [ORNL ED #68](https://code.ornl.gov/sns-hfir-scse/diffraction/engineering-diffraction/engineering-diffraction/-/issues/68)

**To test:**

1. Check all tests are passed (no test is about the algorithm that has defect fixed)
2. Manual test
   - Launch Mantid workbench
   - Run `StartLiveData` from algorithm dialog.
   - Select `Starting Time` as 4 or 5 minutes ago.
   - Do arbitrary setup for the others. 
   - Execute
   - Without the fix, there will be an error message complaining about converting an empty string to DateAndTime
   - With the fix, the `StartTime` in `StartLiveData` algorithm will be set to a correct DateAndTime in ISO format.



Fixes [ORNL ED #68](https://code.ornl.gov/sns-hfir-scse/diffraction/engineering-diffraction/engineering-diffraction/-/issues/68)





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
